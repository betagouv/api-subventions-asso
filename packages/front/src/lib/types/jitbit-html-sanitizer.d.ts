declare module "jitbit-html-sanitizer";
