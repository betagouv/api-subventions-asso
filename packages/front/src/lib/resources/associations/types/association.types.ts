import type {
    BodaccRecordDto,
    ExtraitRcsDto,
    Versement,
    Etablissement,
    DemandeSubvention,
    Siret,
    Siren,
    Rna,
    Adresse,
} from "dto";

export interface Association {
    siren?: Siren;
    rna?: Rna;
    nic_siege?: string;
    categorie_juridique?: string;
    denomination_siren?: string;
    denomination_rna?: string;
    date_creation_siren?: Date;
    date_creation_rna?: Date;
    date_modification_rna?: Date;
    date_modification_siren?: Date;
    objet_social?: string;
    code_objet_social_1?: string;
    code_objet_social_2?: string;
    etablisements_siret?: Siret[];
    adresse_siege_rna?: Adresse;
    adresse_siege_siren?: Adresse;
    federation?: string;
    licencies?: {
        // Uniquement les asso sportive
        total?: number;
        hommes?: number;
        femmes?: number;
    };
    benevoles?: {
        nombre?: number;
        ETPT?: number;
    };
    salaries?: {
        nombre?: number;
        cdi?: number;
        cdiETPT?: number;
        cdd?: number;
        cddETPT?: number;
        emploisAides?: number;
        emploisAidesETPT?: number;
    };
    volontaires?: {
        nombre?: number;
        ETPT?: number;
    };
    versements?: Versement[];
    etablissements?: ({ demandes_subventions: DemandeSubvention[] | null } & Etablissement)[] | null;
    extrait_rcs?: ExtraitRcsDto | null;
    bodacc?: BodaccRecordDto[];
}
