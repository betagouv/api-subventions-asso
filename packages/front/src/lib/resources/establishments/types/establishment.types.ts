import type { Adresse } from "dto/shared/Adresse";
import type { Siret } from "dto/shared/Siret";

export interface SimplifiedEstablishment {
    siret: Siret;
    nic: string;
    siege?: boolean;
    ouvert?: boolean;
    adresse?: Adresse;
    headcount?: string;
}
