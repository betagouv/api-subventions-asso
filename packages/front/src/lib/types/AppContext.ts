export interface AppContext {
    getEnv: "";
    getName: "";
    getDescription: "";
    getContact: "";
    getRepo: "";
}
