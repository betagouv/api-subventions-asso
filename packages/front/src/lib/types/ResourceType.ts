export type ResourceType = "association" | "establishment";
