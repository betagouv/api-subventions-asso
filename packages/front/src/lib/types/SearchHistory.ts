import type { Adresse, RnaDto, SirenDto } from "dto";

export interface SearchHistory {
    rna: RnaDto;
    siren: SirenDto;
    name: string;
    address: Adresse;
    nbEtabs: number;
}
