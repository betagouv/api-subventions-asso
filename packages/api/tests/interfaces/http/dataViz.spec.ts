import amountsVsProgramRegionPort from "../../../src/dataProviders/db/dataViz/amountVSProgramRegion/amountsVsProgramRegion.port";
import { AMOUNTS_VS_PROGRAM_REGION_ENTITIES } from "../../../src/modules/dataViz/amountsVsProgramRegion/__fixtures__/amountsVSProgramRegion.fixture";
import request from "supertest";
import { createAndGetUserToken } from "../../__helpers__/tokenHelper";

jest.mock("../../../src/modules/provider-request/providerRequest.service");

import { App } from "supertest/types";

const g = global as unknown as { app: App };

const insertData = async () => {
    await amountsVsProgramRegionPort.upsertMany(AMOUNTS_VS_PROGRAM_REGION_ENTITIES);
};

describe("DataVizHttp", () => {
    beforeEach(async () => {
        await insertData();
    });

    describe("GET /dataviz/montant-versus-programme-region", () => {
        it("should return amounts versus program region data", async () => {
            const response = await request(g.app)
                .get("/dataviz/montant-versus-programme-region")
                .set("x-access-token", await createAndGetUserToken())
                .set("Accept", "application/json");
            expect(response.status).toBe(200);
            expect(response.body).toMatchSnapshot();
        });
    });
});
