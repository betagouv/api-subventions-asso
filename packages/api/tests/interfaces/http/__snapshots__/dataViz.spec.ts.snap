// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataVizHttp GET /dataviz/montant-versus-programme-region should return amounts versus program region data 1`] = `
Object {
  "amountsVersusProgramRegionData": Array [
    Object {
      "exerciceBudgetaire": 2023,
      "mission": "Mission Exemple",
      "montant": 8000,
      "programme": "1 - Programme Exemple",
      "regionAttachementComptable": "Bretagne",
    },
    Object {
      "exerciceBudgetaire": 2023,
      "mission": "Mission Exemple",
      "montant": 1000,
      "programme": "1",
      "regionAttachementComptable": "Normandie",
    },
    Object {
      "exerciceBudgetaire": 2025,
      "mission": "Mission Exemple",
      "montant": 1000,
      "programme": "1 - Programme Exemple",
      "regionAttachementComptable": "Bretagne",
    },
    Object {
      "exerciceBudgetaire": 2023,
      "mission": "Mission Exemple",
      "montant": 1000,
      "programme": "programNumber",
      "regionAttachementComptable": "Bretagne",
    },
    Object {
      "exerciceBudgetaire": 2023,
      "mission": "Mission Exemple",
      "montant": 1000,
      "programme": "1 - Programme Exemple",
      "regionAttachementComptable": "Occitanie",
    },
  ],
}
`;
